app/javascript/__tests__/modules/listingDetails/ListingDetailsApplicationDate.test.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react"
+import renderer from "react-test-renderer"
+import { ListingDetailsApplicationDate } from "../../../modules/listingDetailsAside/ListingDetailsApplicationDate"
+
+describe("Listing Details Application Date", () => {
+  it("displays Application Deadline when due date has not passed", () => {
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
```

Comment:
We may eventually want to create some json files in the testing dir for common data samples

Reply:
I think as an alternative to `any` here we could consider a partial type: https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
Cool thanks! Didn't know about partials.

Looks like we would need to change the param type on the actual component too to a Partial. I don't think we'd want to do that to accommodate a test. I'll look into passing in full samples of the listing objects in subsequent ticket